azure-pipelines.yml

```diff
@@ -36,23 +36,23 @@ jobs:
           PYTHON_ARCH: 'x64'
           BUILD_TYPE: 'normal'
           imageName: 'windows-2019'
-        Win-Python39-64bit-full-wheel:
-          PYTHON_VERSION: '3.9'
+        Win-Python310-64bit-full-wheel:
+          PYTHON_VERSION: '3.10'
           PYTHON_ARCH: 'x64'
           BUILD_TYPE: 'wheel'
-          NUMPY_MIN: '1.20.0'
+          NUMPY_MIN: '1.21.6'
```

Comment:
@tylerjereddy I was hoping I might be able to pick your brains on this one.

Essentially for the python 3.10 "full wheels" runners (i.e. no deps pip install of MDA, then install deps, then install the minimum version of numpy), the linux runner works perfectly fine when setting `NUMPY_MIN` to `1.21.6` (the version required to build MDAnalysis for python 3.10). However, the windows one ends up with a ton of ABI issues when running tests. 

I can't seem to work out why this is happening. My understanding is that since `NUMPY_MIN` is equivalent to the numpy version using to build MDAnalysis then there should be no ABI compatibility issues?

Reply:
@IAlibay I think because we depend on SciPy (wheels), which have a strict Windows Python 3.10 requirement for newer NumPy because of https://github.com/scipy/scipy/blob/main/pyproject.toml#L38.

```
# On Windows we need to avoid 1.21.6, 1.22.0 and 1.22.1 because they were
# built with vc142. 1.22.3 is the first version that has 32-bit Windows
# wheels *and* was built with vc141. So use that:
"numpy==1.22.3; python_version=='3.10' and platform_system=='Windows' and platform_python_implementation != 'PyPy'",
```

So the transitive dependency on newer NumPy creates an ABI mismatch when using SciPy functionality. i.e.,

```
<snip>
C:\hostedtoolcache\windows\Python\3.10.8\x64\lib\site-packages\scipy\spatial\_kdtree.py:5: in <module>
    from ._ckdtree import cKDTree, cKDTreeNode
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
```

Reply:
Hopefully the MDA team can live with bumping up the min to accommodate that?

Reply:
Alternatively, I suppose pin to an older SciPy, not sure how much luck you'd have with that, but suppose it may be an option perhaps.

Reply:
Aha! Thanks so much @tylerjereddy!

@MDAnalysis/coredevs would you be ok with us just adding an extra pyproject.toml line to bump up the windows numpy requirements by one minor version?

Reply:
Sounds good to me 🤗